### DISCUSSION TITLE
Element Custom Sources saved on user instead of yaml

### DISCUSSION THREAD

**Discussion by peirix** [craftcms/cms#11355] [Ideas] [answered]
It just doesn't make that much sense to save the custom sources in the yaml file. We have a site with multiple editors, each having their own responsible areas, and it would make much more sense, imo, to let them create their own custom sources based on their own needs instead of me creating 1-2 sources per user and enabling/disabling based on which user is logged in. The way it's set up now, they can't even create their own sources on the fly, or change which columns are showing in the list (which was at least possible to do in Craft 3).

**Comment by brandonkelly** [comment]
> The way it's set up now, they can't even create their own sources on the fly, or change which columns are showing in the list (which was at least possible to do in Craft 3).

In Craft 3, only admins could change the visible columns, and they would be changing them for all users, not just themselves. So the only thing that has changed in v4 is where the (still global) settings are stored.

We do plan on adding the ability for users to save their own custom element filters (#11215), and allowing users to customize the visible table columns would make a lot of sense too.

**Reply by peirix** [reply]
But at least I could let the admin users do these actual changes themselves on a live site back in Craft 3. Now I'll have to set up these things for them in dev and then push that to live, removing the functionality of them testing out what works, and changing it around on a day-to-day basis. It's not a major pain, but still a bit of a pain, especially considering Craft 4's focus on AX. :/

**Reply by peirix** [reply]
Bringing back the permission in Craft 3 called «Customize element sources» would be great.

**Reply by brandonkelly** [reply]
To be clear, we aren’t going to revert the move to storing these settings in the project config. But we can certainly look into giving users the ability to override the default table columns for themselves.

**Reply by mmikkel** [reply]
It’d be great to be able to have it both ways. FWIW, storing it in PC makes a ton of sense on our end, as we never allow clients to customize the sources.

**Reply by rynpsc** [reply]
I'm in support of making it optional should it be made customisable per user or some sort of hybrid. One thing we have struggled with in the past is clients seeing different things compared to their colleagues as we've either configured a different view for them or their colleague has forgotten they've changed it themselves. So in some cases it's nice to have a single source of truth. On the other hand some users are more "power users" or as an admin I might want to see more info such as ID columns but don't want that finding its way into production via the Project Config.

**Comment by sunscreem** [comment]
Had a client get in touch this morning saying they pretty annoyed that their team aren't able to customise the columns per user like before. Would be great to see this functionality return.

**Reply by brandonkelly** [reply]
It was never per-user.

**Comment by brandonkelly** [comment, marked as answer]
Element indexes are going to have a new “View” menu in Craft 4.3, which provides a way for users to customize the visible table columns, without affecting other users. (#11915)

<img width="455" alt="The new View menu for an entry index, showing that the view is configured to sort entries by their post date in descending order, and there are six table columns selected. There are “Use defaults” and “Close” buttons at the bottom of the menu." src="https://user-images.githubusercontent.com/47792/190020935-15d3d00c-14b4-4810-a47f-332c3e0f0445.png">

**Reply by brandonkelly** [reply]
Craft 4.3 was released earlier this week, with this change!

**Reply by owldesign** [reply]
Is there a way to "reset" or delete the custom element source selection. I'm trying got go back to what the plugin sets for custom element sources. But once you custom created "headings" or rearange the sources I can't see a way to reset it to get back to original.

**Reply by brandonkelly** [reply]
@owldesign It’s expected you’re going to want to fine-tune the sources to some degree, so there’s no way to reset them in the UI, but you can run this command to do it:

```sh
php craft project-config/remove elementSources.craft\\elements\\Entry
```

**Reply by owldesign** [reply]
This worked, thank you!

-> I had to click reapply everything for project config for this to take effect
